Client/Frontend/Reader/ReadabilityService.swift

```diff
@@ -64,13 +64,16 @@ class ReadabilityOperation: NSOperation, WKNavigationDelegate, ReadabilityBrowse
                 break
             case .Success(let readabilityResult):
                 var error: NSError? = nil
-                if !ReaderModeCache.sharedInstance.put(url, readabilityResult, error: &error) {
+                do {
+                    try ReaderModeCache.sharedInstance.put(url, readabilityResult)
+                } catch let error1 as NSError {
+                    error = error1
                     if error != nil {
```

Comment:
Also we don't need the if statement since the error we get back from the catch isn't optional